pkg/sourcesystem/sourceinterface.go

```diff
@@ -1 +1,55 @@
 package sourcesystem
+
+import (
+	"os"
+
+	"github.com/Gamebuildr/Gogeta/pkg/config"
+	uuid "github.com/satori/go.uuid"
+)
+
+// VersionControl is the interface for specific
+// version control integrations
+type VersionControl interface {
+	CloneSource(repo *SourceRepository, location string) error
+	PullSource() error
+}
+
+// SourceControlManager is the main system for
+// running source control operations
+type SourceControlManager struct {
+	VersionControl VersionControl
+}
+
+// SystemSCM is a SCM that saves repositories
+// locally on the file system
+type SystemSCM SourceControlManager
```

Comment:
The idea here is that in the interface we can specify system level features for the source saving process. The SystemSCM saves a file directly to the local system - also thinking of having one that can stream the clone straight to google cloud (or clone and remove). Will be expanding this in the future.